File: labs/m5-agent-sdk/python/solution/bughunter/schema.py
"""Findings schema — same shape as labs/shared/findings.schema.json (M4) and Claude Security JSONL minus cwe_id (M7)."""
FINDING = {
    "type": "object",
    "properties": {
        "id": {"type": "string", "description": "F1, F2, ..."},
        "title": {"type": "string"},
        "severity": {"type": "string", "enum": ["HIGH", "MEDIUM", "LOW"]},
        "file": {"type": "string", "description": "path relative to the repository root"},
        "line": {"type": "integer", "minimum": 1},
        "category": {"type": "string", "description": "e.g. error-handling, concurrency, input-validation, resource-leak, security"},
        "description": {"type": "string"},
        "recommendation": {"type": "string"},
        "confidence": {"type": "string", "enum": ["low", "medium", "high"]},
    },
    "required": ["id", "title", "severity", "file", "line", "category", "description", "recommendation", "confidence"],
    "additionalProperties": False,
}
FINDINGS_SCHEMA = {
    "type": "object",
    "properties": {
        "service": {"type": "string"},
        "summary": {"type": "string"},
        "findings": {"type": "array", "items": FINDING},
    },
    "required": ["service", "summary", "findings"],
    "additionalProperties": False,
}

File: labs/m5-agent-sdk/typescript/solution/src/schema.ts
import { z } from "zod";

export const Finding = z.object({
  id: z.string(), title: z.string(), severity: z.enum(["HIGH", "MEDIUM", "LOW"]),
  file: z.string(), line: z.number().int().min(1), category: z.string(),
  description: z.string(), recommendation: z.string(), confidence: z.enum(["low", "medium", "high"]),
});
export const Findings = z.object({ service: z.string(), summary: z.string(), findings: z.array(Finding) });
export type Findings = z.infer<typeof Findings>;
// The SDK validates against JSON Schema draft-07; Zod 4 emits 2020-12 unless told otherwise.
export const FINDINGS_JSON_SCHEMA = z.toJSONSchema(Findings, { target: "draft-7" }) as Record<string, unknown>;
